### PR TITLE
Motion primitives unit tests

### PR DESCRIPTION
**Pull request by orduno**
This PR adds some unit tests for the `MotionPrimitive` base class. Addressing #384.

In the process of making the tests, I found a need for adding a stop method to the task server base class.

### CODE REVIEW

File: nav2_motion_primitives/CMakeLists.txt [hunk 1]
```diff
@@ -70,7 +70,11 @@ install(DIRECTORY include/
 
 if(BUILD_TESTING)
   find_package(ament_lint_auto REQUIRED)
+  find_package(ament_cmake_gtest REQUIRED)
+
   ament_lint_auto_find_test_dependencies()
+  ament_add_gtest(test_motion_primitives test/test_motion_primitives.cpp)
```
Comment (crdelsey) [inline nav2_motion_primitives/CMakeLists.txt]: Our convention is to do an `add_subdirectory(test)` here and then make a CMakeLists.txt in the test directory that has the details of the tests ie. the `ament_add_gtest` line and the `ament_target_dependencies` line.

Comment (orduno) [reply nav2_motion_primitives/CMakeLists.txt]: Pushed a change.

**Comment by orduno** [comment]
@crdelsey I've addressed your feedback.

File: nav2_tasks/include/nav2_tasks/task_server.hpp [hunk 2]
```diff
@@ -185,14 +202,15 @@ class TaskServer
           throw std::logic_error("Unexpected status return from task");
         }
       }
-    } while (rclcpp::ok());
+    } while (rclcpp::ok() && !stop_thread_);
   }
 
   // TODO(mjeronimo): Make explicit start and stop calls to control
```
Comment (mjeronimo) [inline nav2_tasks/include/nav2_tasks/task_server.hpp]: You can get rid if this comment now that you've implemented it.

**Pull request merged**